### PR TITLE
Guideline Compliant SS13 Tetris

### PR DESCRIPTION
**Pull request by silicons**
## About The Pull Request

https://tetris.wiki/Tetris_Guideline 

Using that as a design document, attempts to create a relatively performant server side implementation of tetris, rendered using a BYOND secondary map view rather than HTML.
Intended to have PVP modes much like this meme https://jstris.jezevec10.com/ <--- which a bunch of us were playing

Let's be honest this is just a rash PR made while I'm half asleep.

## Why It's Good For The Game

Tetris

## Changelog
:cl:
add: Tetris
/:cl:

### CODE REVIEW

**Comment by Hatterhat** [comment]
OH BABY

**Comment by Linzolle** [comment]
on god

**Comment by MrJWhit** [comment]
I can't wait for the day, that someone comes into this server, goes "Oh, this is the dreaded furry erp server, I'll just join as a scientist so I don't have to talk anyone", and the first thing their asked is how good they are at tetris.

File: code/modules/tetris/_tetris_defines.dm [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#define TETRIS_LOCK_INFINITY		1				//can stall forever by turning
+#define TETRIS_LOCK_LIMITED			2				//can turn <x> number of times to stall
+#define TETRIS_LOCK_STEP			3				//can only delay by making piece move down naturally
+
+#define TETRIS_FIELD_WIDTH					10
+#define TETRIS_FIELD_HEIGHT					40
+#define TETRIS_FIELD_HEIGHT_PLACEMENT		22
+#define TETRIS_FIELD_HEIGHT_SHOW			21
+
+#define TETRIS_PIECE_BAG_DEPTH			7
```
Comment (Ghommie) [inline code/modules/tetris/_tetris_defines.dm:10]: Guideline says these should be called "tetriminos", not "pieces".

File: code/modules/tetris/field.dm [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	var/lock_delay = TETRIS_DEFAULT_LOCK_DELAY
+
+	//actual playing field
+	var/field_width = TETRIS_FIELD_WIDTH
+	var/field_height = TETRIS_FIELD_HEIGHT
+	var/field_placement_height = TETRIS_FIELD_HEIGHT_PLACEMENT
+	var/field_visible_height = TETRIS_FIELD_HEIGHT_SHOW
+	var/list/playing_field						//[row][column] =
+	var/list/datum/tetris_piece/pieces			//pieces on the board
+	var/list/datum/tetris_piece/bag				//pieces in bag
```
Comment (Ghommie) [inline code/modules/tetris/field.dm:19]: .

File: code/modules/tetris/field.dm [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		var/list/L = playing_field[i]
+		L.len = field_width
+
+
+
+
+
+
+
+/datum/tetris_field/proc/get_piece(row, column)
```
Comment (Ghommie) [inline code/modules/tetris/field.dm:48]: .

**Review by Ghommie — COMMENTED** [review]
Good luck.

File: code/modules/tetris/_tetris_defines.dm (hunk 1 above)
Comment (Hatterhat) [reply code/modules/tetris/_tetris_defines.dm:10]: *tetrominoes

Comment (Ghommie) [reply code/modules/tetris/_tetris_defines.dm:10]: Damnit.

**Pull request closed without merging**